hail/python/hailtop/batch/genetics/regenie/regenie.py

```diff
@@ -0,0 +1,310 @@
+    j1.cpu(args.cores)
+    j1.memory(args.memory)
+    j1.storage(args.storage)
+
+    in_step1 = get_input(batch, step1_args)
+
+    phenos = get_phenos(step1_args)
+    nphenos = len(phenos)
+
+    step1_output_prefix = step1_args.out
+    s1out = {"log": f"{step1_output_prefix}.log", "pred_list": f"{step1_output_prefix}_pred.list"}
+
+    for i in range(1, nphenos + 1):
+        s1out[f"{step1_output_prefix}_{i}"] = f"{step1_output_prefix}_{i}.loco"
+
+    if step1_args.lowmem:
+        for i in range(1, nphenos + 1):
+            pfile = f"{step1_args.lowmem_prefix}_l0_Y{i}"
+            s1out[pfile] = pfile
+
+    j1.declare_resource_group(**{step1_output_prefix: s1out})
+
+    cmd1 = []
+    for name, val in vars(step1_args).items():
+        if val is None or val is False or name == "step":
+            continue
+
+        if name in from_underscore:
+            name = from_underscore[name]
+
+        if name in input_file_args:
+            cmd1.append(f"--{name} {in_step1[name]}")
+        elif name == "out":
+            cmd1.append(f"--{name} {j1[step1_output_prefix]}")
+        elif isinstance(val, bool):
+            cmd1.append(f"--{name}")
+        elif name == "phenoCol":
+            for pheno in val:
+                cmd1.append(f"--{name} {pheno}")
+        else:
+            cmd1.append(f"--{name} {val}")
+
+    cmd1 = f"--step 1 {' '.join(cmd1)}"
+
+    j1.command(f"regenie {cmd1}")
+
+    phenos = get_phenos(step2_args)
+    nphenos = len(phenos)
+
+    j2 = batch.new_job(name='run-regenie')
+    j2.image('akotlar/regenie:latest')
+    j2.cpu(args.cores)
+    j2.memory(args.memory)
+    j2.storage(args.storage)
+
+    step2_output_prefix = step2_args.out
+    s2out = {"log": f"{step2_output_prefix}.log"}
+
+    if step2_args.split:
+        for pheno in phenos:
+            out = f"{step2_output_prefix}_{pheno}.regenie"
+            s2out[out] = out
+
+    print(f"Regenie Step 2 output files: \n{s2out.values()}")
+
+    j2.declare_resource_group(**{step2_output_prefix: s2out})
+
+    in_step2 = get_input(batch, step2_args)
+
+    cmd2 = []
+    for name, val in vars(step2_args).items():
+        if val is None or val is False or name == "step" or name == "pred":
+            continue
+
+        if name in from_underscore:
+            name = from_underscore[name]
+
+        if name in input_file_args:
+            cmd2.append(f"--{name} {in_step2[name]}")
+        elif name == "out":
+            cmd2.append(f"--{name} {j2[step2_output_prefix]}")
+        elif isinstance(val, bool):
+            cmd2.append(f"--{name}")
+        elif name == "phenoCol":
+            for pheno in val:
+                cmd2.append(f"--{name} {pheno}")
+        else:
+            cmd2.append(f"--{name} {val}")
+
+    if not step2_args.ignore_pred:
+        cmd2.append(f"--pred {j1[step1_output_prefix]['pred_list']}")
+
+    cmd2 = f"--step 2 {' '.join(cmd2)}"
+
+    j2.command(f"regenie {cmd2}")
+
+    return j1, j2, step2_output_prefix
```

Comment:
The file names, I think, in your proposal won't be right. I want the output prefix that the user specifies to be respected, and have the final output able to be written to a folder with the name unmodified. This will take a small followup pr.